src_c/simd_blitters_sse2.c

```diff
@@ -490,21 +479,30 @@ alphablit_alpha_sse2_argb_no_surf_alpha_opaque_dst(SDL_BlitInfo *info)
                     /* src(ARGB) -> src1 (00000000ARGBARGB) */
                     LOAD_64_INTO_M128(srcp64, &src1);
 
-                    /* created squashed alpha -> mm_src_alpha
-                     * (0000000000000A0A) */
+                    /* isolate alpha channels
+                     * 00000000A1000A2000 -> mm_src_alpha */
+                    mm_src_alpha = _mm_andnot_si128(mm_rgb_mask, src1);
+
+                    /* shift right to position alpha channels for manipulation
+                     * 000000000A1000A200 -> mm_src_alpha*/
+                    mm_src_alpha = _mm_srli_si128(mm_src_alpha, 1);
+
+                    /* shuffle alpha channels to duplicate 16 bit pairs
+                     * shuffle (3, 3, 1, 1) (backed 2 bit numbers)
+                     * [00][00][00][00][0A1][00][0A2][00] -> mm_src_alpha
+                     * [7 ][6 ][5 ][4 ][ 3 ][2 ][ 1 ][0 ]
+                     * Therefore the previous contents of 16 bit number #1
+                     * Goes into 16 bit number #1 and #2, and the previous
+                     * content of 16 bit number #3 goes into #2 and #3 */
                     mm_src_alpha =
-                        _mm_add_epi16(_mm_and_si128(_mm_srli_si128(src1, 3),
-                                                    mm_alpha_mask_1),
-                                      _mm_and_si128(_mm_srli_si128(src1, 5),
-                                                    mm_alpha_mask_2));
+                        _mm_shufflelo_epi16(mm_src_alpha, 0b11110101);
```

Comment:
These are very helpful comments. Thanks.

I wonder if either of you have good docs in your code editor for these intrinsics?

It's possible to look them up, but would be quicker if it's shown in the editor. https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_shufflelo_epi16&ig_expand=6448

Reply:
Occasionally I see short text descriptions of the intrinsics in VS code (pulled from the header I believe), but mainly I just rely on the intel intrinsics guide for these things.